### DISCUSSION TITLE
[19.11.2024] Blog 6 - Backlog

### DISCUSSION THREAD

**Discussion by TimHitschfeld** [bunki1224/Softwareengineering-Projekt#46] [Show and tell]
In dieser Woche haben wir weiter an unserer TripAhead App in React gearbeitet. Nachdem wir letzte Woche eine leicht anpassbare Komponente für die Timeline-Ansicht entwickelt haben, ging es diese Woche darum, diese mit Inhalt zu befüllen. Zuerst haben wir eine Ansicht gestaltet, welche aus einer Backlog und einer Timeline spalte besteht.

![image](https://github.com/user-attachments/assets/86c29af6-db64-4ddc-bd2f-69a0837d29b4)

Zusätzlich besitzt die Backlogspalte eine Suchleiste. Mit dieser kann man nach Google Maps Elementen Suchen und diese durch einen Klick oder die Entertaste als Komponente zum Backlog hinzufügen.

![image](https://github.com/user-attachments/assets/69572bc3-18e3-40a6-91cf-26b9af80747a)

Danach kann man den Inhalt der Komponente beliebig anpassen.

![image](https://github.com/user-attachments/assets/64ed5991-a1d5-4602-a6b8-5301e0e0a2de)

Für nächste Woche nehmen wir uns vor, Drag und Drop hinzuzufügen um die Komponenten in die Timeline zu schieben und die Funktion die Komponenten zu löschen.

Euer TripAhead Team /TH

**Comment by Ninetilt** [comment]
Die Ansicht sieht schon mal sehr clean und intuitiv aus. Man erkennt sofort die wichtige Funktionen und ist nicht überwältigt mit einer Masse an Buttons und Symbolen. Die tatsächliche Einbindung der Google Maps API finden wir bemerkenswert. Vorher haben wir diese Funktion ja nur theoretisch in euren Verhaltensmodellen sehen können. Die Individualisierbarkeit bereits gespeicherter Activities kommt uns auch sehr nützlich vor. Drag and Drop ist definitiv eine sehr wichtige Funktion, da die Oberfläche selbst schon so aussieht, als könnte man sie mit Drag and Drop bedienen.

Im Großen und Ganzen ein guter Fortschritt 🙂 

Viele Grüße
Projekt TDC (JS)

**Comment by DonKir15** [comment]
Die Ansicht sieht richtig gut durchdacht aus und wirkt super übersichtlich. Man findet sich direkt zurecht, und die Funktionen sind klar erkennbar, ohne dass es zu überladen wirkt. Die Google Maps-Suche ist ein richtig cooler Touch . Auch die Möglichkeit, die Aktivitäten später noch anzupassen, ist mega praktisch! Drag-and-Drop passt perfekt zu eurer Oberfläche. Damit wird das Ganze noch intuitiver und macht sicher auch mehr Spaß beim Planen. Bin echt gespannt, wie das mit der Timeline am Ende aussieht! Weiter so!

**Comment by schmid-di** [comment]
Das klingt nach spannender Arbeit an eurer TripAhead App! Besonders beeindruckt mich, wie ihr die Timeline-Ansicht nicht nur optisch, sondern auch funktional entwickelt. Die Kombination aus einer Backlog-Spalte mit Suchleiste und die Möglichkeit, Inhalte direkt aus Google Maps hinzuzufügen, klingt äußerst nutzerfreundlich und intuitiv.

Euer Fokus auf Flexibilität – sowohl bei der Anpassung der Inhalte als auch in der Planung für Drag-and-Drop-Funktionalitäten – zeigt, dass ihr eine echte „User First“-Mentalität verfolgt. Das wird die Usability eurer App sicherlich stark verbessern.

Ich bin gespannt, wie ihr das Drag-and-Drop-Feature und die Löschfunktion implementieren werdet. Nutzt ihr dafür eine Library wie react-beautiful-dnd, oder entwickelt ihr eine eigene Lösung? Viel Erfolg bei der weiteren Umsetzung – ich freue mich darauf, mehr über den Fortschritt zu lesen! 😊

Euer UMOC-Team 

/JMS /PL /ML /MR

**Comment by oskarkreutzer** [comment]
Das Konzept der Backlog-Spalte mit Suchleiste wirkt äußerst durchdacht und nutzerfreundlich. Besonders spannend finden wir die Integration der Google Maps API. Eine Frage hätten wir: Wie plant ihr, die Benutzerfreundlichkeit beim Löschen und Verschieben von Komponenten sicherzustellen, um versehentliche Aktionen zu vermeiden? Z. B. durch eine Rückgängig-Funktion?

Euer GymTrackerX Team
